File: src/Models/Invoice.ts
import {Column, CreateDateColumn, Entity, ManyToOne, PrimaryColumn, UpdateDateColumn} from "typeorm";
import {BusinessOwner} from "./BusinessOwner";




@Entity()
export class Invoice {
    @PrimaryColumn()
    invoiceId: number;
    @Column()
    invoiceType: string;
    @Column()
    invoiceTotal: number;
    @Column()
    invoiceDate: string;
    @Column()
    invoiceStatus: string;
    @Column()
    invoiceAssetUrl: string;
    @ManyToOne(() => BusinessOwner, businessOwner => businessOwner.invoices)
    businessOwner: BusinessOwner;
    @CreateDateColumn()
    createdAt: Date;
    @UpdateDateColumn()
    updatedAt: Date;

}

File: src/main/java/iewa/api/Model/Invoice.java
package iewa.api.Model;


import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "invoice")
@Data
public class Invoice {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private String invoiceId;

    @Column
    private String invoiceType;

    @Column
    private String invoiceDate;

    @Column
    private String invoiceMondayId;

    @Column
    private String invoiceStatus;

    @Column
    private String invoiceAssetUrl;

    @Column
    private String invoiceAmount;

    @ManyToOne(fetch = FetchType.EAGER)
    @JsonBackReference
    private BusinessOwner businessOwner;


}
